File: event-service/src/main/java/org/tggc/eventservice/service/EventService.java
package org.tggc.eventservice.service;

import org.tggc.eventservice.dto.EventRq;
import org.tggc.eventservice.dto.EventRs;
import org.tggc.userapi.dto.UserDto;

import java.time.LocalDateTime;
import java.util.List;

public interface EventService {

    EventRs getEventById(Long eventId);

    List<EventRs> getEventsByUser(Long userId);

    EventRs createEvent(EventRq event, Long userId);

    void joinEvent(Long eventId, Long userId);

    void deleteEvent(Long eventId);

    List<UserDto> getUsersByEvent(Long eventId);

    void leaveEvent(Long eventId, Long userId);

    List<EventRs> getEventsByFilter(String title,
                                    LocalDateTime startDate,
                                    LocalDateTime endDate,
                                    Long creatorId);
}

File: event-service/src/main/kotlin/org/tggc/eventservice/service/EventService.kt
package org.tggc.eventservice.service

import org.tggc.eventservice.dto.EventRq
import org.tggc.eventservice.dto.EventRs
import org.tggc.userapi.dto.UserDto
import java.time.LocalDateTime

interface EventService {

    fun getEventById(eventId: Long): EventRs

    fun getEventsByUser(userId: Long): List<EventRs>

    fun createEvent(rq: EventRq, userId: Long): EventRs

    fun joinEvent(eventId: Long, userId: Long)

    fun deleteEvent(eventId: Long)

    fun getUsersByEvent(eventId: Long): List<UserDto>

    fun leaveEvent(eventId: Long, userId: Long)

    fun getEventsByFilter(
        title: String?,
        startDate: LocalDateTime?,
        endDate: LocalDateTime?,
        creatorId: Long?
    ): List<EventRs?>
}
